lib/run/options.js

```diff
@@ -267,25 +267,17 @@ var _ = require('lodash'),
         iterationData: function (location, options, callback) {
             if (_.isArray(location)) { return callback(null, location); }
 
-            util.fetch(location, function (err, data) {
+            util.fetch(location, function (err, data, fileType) {
                 if (err) {
                     return callback(new Error(ITERATION_DATA_LOAD_ERROR_MESSAGE + `\n  ${err.message || err}`));
                 }
 
-                // Try loading as a JSON, fall-back to CSV. @todo: switch to file extension based loading.
+                // Check if the file is JSON else parse as CSV
                 async.waterfall([
                     (cb) => {
-                        try {
+                        if (fileType === '.json') {
                             return cb(null, liquidJSON.parse(data.trim()));
```

Comment:
@elit-altum I think here `try...catch` should be there to catch .json parsing errors.

Reply:
This was skipped as the previous ```try-catch``` code was just for catching the is JSON or not error. Therefore, for an invalid JSON the error message provided by newman was 
```
error: iteration data could not be loaded
 Invalid Closing Quote: found non trimable byte after quote at line 2
```

this was because ```parseCSV``` was handling the JSON error.